pkg/server/worker.go

```diff
@@ -29,41 +41,41 @@ func (s *Server) worker(name string, h *host.Host) {
 			s.logger.Infof("Worker for host %s: Latency=%v (Ping successful)", name, latency)
 			h.Alive = true
 			h.Latency = latency
-			// Update the RRD file with the fetched latency and timestamp
-			s.logger.Debugf("Worker for host %s: Updating RRD with latency %f microseconds.", name, float64(latency.Microseconds()))
-			err = rrdFile.SafeUpdate(time.Now(), []float64{float64(latency.Microseconds())})
-			if err != nil {
-				s.logger.Errorf("Worker for host %s: Failed to update RRD (%v)", name, err)
-			} else {
-				s.logger.Debugf("Worker for host %s: RRD update successful.", name)
-			}
+			latencyUpdate = []float64{float64(latency.Microseconds())}
+		}
+		// Update the RRD file with the fetched latency and timestamp
+		// If the slice is empty, SafeUpdate will handle it
```

Comment:
this isn't great and `SafeUpdate` will need to be cut up into smaller functions in the future